crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -966,6 +966,33 @@ impl Context {
         result_ids: &[ValueId],
     ) -> Result<Vec<AcirValue>, RuntimeError> {
         match intrinsic {
+            Intrinsic::AssertEq => {
+                let lhs = self.convert_value(arguments[0], dfg);
+                let rhs = self.convert_value(arguments[1], dfg);
+
+                fn get_var_equality_assertions(
+                    lhs: AcirValue,
+                    rhs: AcirValue,
+                ) -> Vec<(AcirVar, AcirVar)> {
+                    match (lhs, rhs) {
+                        (AcirValue::Var(lhs, _), AcirValue::Var(rhs, _)) => vec![(lhs, rhs)],
+                        (AcirValue::Array(lhs_values), AcirValue::Array(rhs_values)) => lhs_values
+                            .into_iter()
+                            .zip(rhs_values)
+                            .flat_map(|(lhs, rhs)| get_var_equality_assertions(lhs, rhs))
+                            .collect(),
+
+                        // TODO: How should I handle `AcirValue::DynamicArray`?
```

Comment:
https://github.com/noir-lang/noir/blob/3d1b2522c8d9a0acebff102f9f877c44178c5db5/crates/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs#L902-L917

Reply:
https://github.com/noir-lang/noir/blob/b6c36aa224c99641c8b0dbe9e49cd4124e8c0f33/crates/noirc_evaluator/src/ssa/acir_gen/mod.rs#L248

Actually, looks like I can ignore this case.